src/Compilers/Shared/BuildServerConnection.cs

```diff
@@ -416,7 +416,19 @@ internal static bool TryCreateServerCore(string clientDir, string pipeName)
             {
                 try
                 {
-                    Process.Start(expectedPath, processArguments);
+                    var startInfo = new ProcessStartInfo()
+                    {
+                        FileName = expectedPath,
+                        Arguments = processArguments,
+                        UseShellExecute = false,
+                        WorkingDirectory = clientDir,
+                        RedirectStandardInput = true,
+                        RedirectStandardOutput = true,
+                        RedirectStandardError = true,
+                        CreateNoWindow = true
```

Comment:
Where are the streams redirected to in this case?

Reply:
On Unix, CoreFX calls into CoreCLR to create an anonymous pipe, then puts the child end in the new process and the parent end in the calling process. It closes the dangling files on both ends. Then it creates a wrapper FileStream for each fd.

This sounds OK except for one scenario: we never read from the output pipes in the caller. In theory this should be fine since we should never be writing to the pipe anyway, but if we ever did write to the pipe in VBCSCompiler we could end up filling up a buffer and blocking, which would be bad, but I'm counting on creating invalid output streams on Windows to catch that.

I looked into immediately trying to close these streams after completing process creation successfully, but couldn't convince myself it was always safe.